forte/data/readers/base_reader.py

```diff
@@ -50,15 +50,15 @@ class BaseReader(PipelineComponent[PackType], ABC):
 
     def __init__(self,
                  from_cache: bool = False,
-                 cache_directory: Optional[Path] = None,
+                 cache_directory: Optional[str] = None,
                  append_to_cache: bool = False):
         """
         Args:
             from_cache (bool, optional): Decide whether to read from cache
-                if cache file exists. By default (``True``), the reader will
-                try to read an datapack from the first line of the caching file.
-                If ``False``, the reader will only read from the original file
-                and use the cache file path only for output.
+                if cache file exists. By default (``False``), the reader will
+                only read from the original file and use the cache file path
+                only for output. If ``True``, the reader will try to read a
```

Comment:
Thanks for fixing the docstring, but I realize the original docstring is not very accurate.

"use the cache file path only for output"?why is the cache file related to output? This parameter should  only affect whether this reader reads from the cache, but it shouldn't affect its behavior on its own caching.

Reply:
I think it meant that the contents of the reader are written to the cache file if the `cache_directory` is not None basically, it is not affecting the caching behaviour but letting the user know that the reader will not read from `cache_directory` but only write to it. Otherwise,  the cache file path(`cache_directory` + whatever `_cache_key_function` returns) is not used. I think the words used are cause of miscommunication. This is the expected behaviour right?

Reply:
Yeah, just wording issues.